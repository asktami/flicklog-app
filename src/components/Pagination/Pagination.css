.pagination,
.pagination li a {
	display: flex;
	flex-wrap: wrap;
	justify-content: center;
	align-items: center;
}

.pagination li {
	border-radius: 4px;
	border: 1px solid #efefef;
	list-style: none;
}

.pagination button {
	color: black;
	float: left;
	padding: 8px 16px;
	text-decoration: none;
	transition: background-color 0.3s;
}

.pagination button.active {
	background-color: #4caf50;
	color: white;
}

.pagination button:hover:not(.active) {
	background-color: #ddd;
}

.pagination li:not([class*='current']) a:focus,
.pagination li:not([class*='current']) a:active {
	box-shadow: 0px 0px 10px 1px rgba(0, 0, 0, 0.25);
	border-left-width: 1px;
}
