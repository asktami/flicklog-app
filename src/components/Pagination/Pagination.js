import React from 'react';
import './Pagination.css';

const Pagination = ({
	handleChangePage,
	currentPage,
	totalPages,
	loadMore,
}) => {
	const pageNumbers = [];

	for (let i = 1; i <= totalPages; i++) {
		pageNumbers.push(i);
	}

	console.log('totalPages = ', totalPages);
	console.log('lastPage = ', pageNumbers.length);
	console.log(JSON.stringify(pageNumbers));

	return (
		<section>
			{/* <ul>
				{totalPages ? (
					<li>
						{currentPage !== 1 ? (
							<button
								onClick={() => handleChangePage(currentPage - 1)}
								className="page-link"
							>
								«
							</button>
						) : (
							'«'
						)}
					</li>
				) : null}
				&nbsp;&nbsp;&nbsp;Page&nbsp;&nbsp;
				<span>{currentPage}</span>
				&nbsp;&nbsp; of &nbsp;&nbsp;
				<span>{totalPages}</span>
				&nbsp;&nbsp;&nbsp;
				{pageNumbers.length ? (
					<li>
						{currentPage !== pageNumbers.length ? (
							<button
								onClick={() => handleChangePage(currentPage + 1)}
								className="page-link"
							>
								»
							</button>
						) : (
							'»'
						)}
					</li>
				) : null}
			</ul> */}

			<ul className="pagination">
				{totalPages && currentPage !== 1 ? (
					<li>
						<button
							onClick={() => handleChangePage(currentPage - 1)}
							className="page-link"
						>
							«
						</button>
					</li>
				) : null}
				{pageNumbers.map((nbr) => (
					<li key={nbr}>
						<button
							onClick={() => handleChangePage(nbr)}
							className={`page-link ${currentPage === nbr ? ' active' : ''}`}
						>
							{nbr}
						</button>
					</li>
				))}
				{pageNumbers.length && currentPage !== pageNumbers.length ? (
					<li>
						<button
							onClick={() => handleChangePage(currentPage + 1)}
							className="page-link"
						>
							»
						</button>
					</li>
				) : null}
			</ul>
		</section>
	);
};

export default Pagination;
