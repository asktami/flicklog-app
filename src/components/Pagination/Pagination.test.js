import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Pagination from './Pagination';

describe(`Pagination component`, () => {
	it('renders a Pagination by default', () => {
		const wrapper = shallow(<Pagination />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
