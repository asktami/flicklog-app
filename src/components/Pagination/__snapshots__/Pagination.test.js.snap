// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component renders a Pagination by default 1`] = `
<section>
  <ul
    className="pagination"
  />
</section>
`;
